src/connection.cpp

```diff
@@ -154,11 +144,13 @@ void Connection::setMAC(std::string_view mac)
     if ((mac.size() != 12) && (mac.size() != 14) && (mac.size() != 17))
         throw std::runtime_error("Invalid MAC address size");
 
-    // TODO: Make it easier to read and consider the Cisco MAC identifier
-    const std::regex pattern("^([0-9A-Fa-f]{2}[:-]){5}"
-                             "([0-9A-Fa-f]{2})|([0-9a-"
-                             "fA-F]{4}\\.[0-9a-fA-F]"
-                             "{4}\\.[0-9a-fA-F]{4})$");
+    const std::regex pattern(
+        R"regex(^
+    ([0-9A-Fa-f]{2}[:-]){5}        # Matches MAC address with colons or hyphens
+    ([0-9A-Fa-f]{2})|              # Or
+    ([0-9a-fA-F]{4}\.[0-9a-fA-F]{4}\.[0-9a-fA-F]{4})  # Matches IPv6 format
```

Comment:
Do you have an example? IPv6?

Reply:
> Matches MAC address with colons or hyphens: Example: 12:34:56:78:9A:BC
> 
> Matches IPv6 format: Example: 1a2b:3c4d:5e6f:7a8b:9c1d:1e2f:3a4b:5c6d
> 
> Matches Cisco MAC format: Example: 12ab.34cd.56ef

Alright but it makes no sense for IPv6. IPv6 is an IP address and not an MAC Address.